.github/workflows/test_terraform_module.yaml

```
@@ -0,0 +1,55 @@
+# Copyright 2025 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+# Attention: valid for K8s charms only
+
+name: Terraform module tests
+
+on:
+  pull_request:
+    paths:
+      - 'terraform/**'
+
+permissions:
+  contents: read
+
+jobs:
+  test-terraform:
+    name: Test Terraform with Juju
+    runs-on: ubuntu-latest
+    env:
+      WORKING_DIR: 'terraform/tests'
+    steps:
+    - uses: actions/checkout@@v4.2.2
+    - uses: charmed-kubernetes/actions-operator@main
+      with:
+        provider: "microk8s"
```

We should start to use Canonical Kubernetes in CI:
```suggestion
        provider: "k8s"
        use-canonical-k8s: true
```